src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -26,65 +26,207 @@ namespace System.Net.Http
-                    ((ExceptionDispatchInfo)info).Throw();
+                    response.Dispose();
+                    response = null;
                 }
 
-                throw;
-            }
+                RTHttpRequestMessage rtRequest = await ConvertRequestAsync(
+                    request,
+                    requestHttpMethod,
+                    skipRequestContentIfPresent).ConfigureAwait(false);
+
+                RTHttpResponseMessage rtResponse;
+                try
+                {
+                    if (redirects > 0)
+                    {
+                        rtResponse = await _filterWithNoCredentials.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        rtResponse = await _filter.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+                    }
+                }
+                catch (TaskCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception)
+                {
+                    object info;
+                    if (rtRequest.Properties.TryGetValue(SavedExceptionDispatchInfoLookupKey, out info))
+                    {
+                        ((ExceptionDispatchInfo)info).Throw();
+                    }
+
+                    throw;
+                }
+
+                response = ConvertResponse(rtResponse);
+
+                ProcessResponseCookies(response, request.RequestUri);
+
+                if (!_handler.AllowAutoRedirect)
+                {
+                    break;
+                }
+
+                if (response.StatusCode != HttpStatusCode.MultipleChoices &&
+                    response.StatusCode != HttpStatusCode.MovedPermanently &&
+                    response.StatusCode != HttpStatusCode.Redirect &&
+                    response.StatusCode != HttpStatusCode.RedirectMethod &&
+                    response.StatusCode != HttpStatusCode.RedirectKeepVerb)
+                {
+                    break;
+                }
+
+                redirects++;
+                if (redirects > _handler.MaxAutomaticRedirections)
+                {
+                    break;
+                }
+
+                Uri redirectUri = response.Headers.Location;
+                if (redirectUri == null)
+                {
+                    break;
+                }
+
+                if (!redirectUri.IsAbsoluteUri)
+                {
+                    redirectUri = new Uri(string.Format("{0}://{1}:{2}{3}",
+                        request.RequestUri.Scheme,
+                        request.RequestUri.Host,
+                        request.RequestUri.Port,
+                        redirectUri.OriginalString));
+                }
+
+                if (redirectUri.Scheme != Uri.UriSchemeHttp &&
+                    redirectUri.Scheme != Uri.UriSchemeHttps)
+                {
+                    break;
```

Comment:
This particular check makes sure that we don't try to redirect to a scheme like FTP or FILE or WS.  So, the only possible destination schemes need to be HTTP or HTTPS.